lib/mix/lib/mix/tasks/test.ex

```diff
@@ -412,8 +412,15 @@ defmodule Mix.Tasks.Test do
             Mix.shell().info("There are no tests to run")
 
           true ->
-            message = "Paths given to \"mix test\" did not match any directory/file: "
-            raise_or_error_at_exit(message <> Enum.join(files, ", "), opts)
+            message =
+              "Paths given to \"mix test\" did not match any directory/file: " <>
+                Enum.join(files, ", ")
+
+            if Mix.ProjectStack.recursing() do
+              Mix.shell().info(message)
+            else
+              raise_or_error_at_exit(message, opts)
```

Comment:
Btw, I think we should add this logic to inside the `raise_or_error_at_exit` function and maybe rename it to `raise_or_error_at_exit_unless_recursing`. The reason is that the same rationale you have for files would apply to `mix test --only foo` and `foo` is only available in one of the umbrella children.